src/components/pages/rules/flyouts/ruleNew/ruleNew.js

```diff
@@ -60,158 +62,248 @@ export class RuleNew extends LinkedComponent {
+    const calculationOptions = [{ value: average, label: average }, { value: instant, label: instant }];
+    this.setState({
+      deviceGroupOptions,
+      calculationOptions
+    });
+  }
+
+  addCondition = () => this.conditionsLink.set([...this.conditionsLink.value, newCondition()]);
 
   deleteCondition = (index) =>
-    () => this.conditions.set(this.conditions.value.filter((_, idx) => index !== idx));
+    (evt) => this.conditionsLink.set(this.conditionsLink.value.filter((_, idx) => index !== idx));
 
   createRule = (event) => {
     event.preventDefault();
     console.log('TODO: Handle the form submission');
   }
 
+  onGroupIdChange = (selectedItem) => {
+    this.getDeviceCountAndFields(selectedItem.target.value.value);
+  }
+
+  getDeviceCountAndFields(groupId) {
+    this.props.deviceGroups.forEach(group => {
+      if (group.id === groupId) {
+        this.subscription = IoTHubManagerService.getDevices(group.Conditions)
+          .subscribe(
+            groupDevices => {
+              this.setState({
+                groupDevices,
+                devicesAffected: groupDevices.length,
+                fieldOptions: this.getConditionFields(groupDevices),
+
+              });
+            },
+            errorResponse => {
+              this.setState({ error: errorResponse.errorMessage });
+            }
+          );
+      }
+    });
+  }
+
+  getConditionFields(devices) {
+    const fields = [];
+    devices.forEach(device => {
+      const telemetry = device.telemetry;
+      if (telemetry) {
+        Object.values(telemetry).forEach(field => {
+          const extract = field.messageSchema.fields;
+          Object.keys(extract).forEach(field => {
+            if (field.indexOf('_unit') !== -1) return; //we don't want keys that contain _
+            if (fields.every(o => o.value !== field)) {
+              fields.push({
+                label: field,
+                value: field
+              });
+            }
+          });
+        });
+      }
+    });
+    return fields;
+  }
+
   render() {
     const { onClose, t } = this.props;
     const name = this.ruleNameLink.forkTo('name')
       .withValidator(ruleNameValidator);
     // Create the state link for the dynamic form elements
     const conditionLinks = this.conditionsLink.getLinkedChildren(conditionLink => {
       const fieldLink = conditionLink.forkTo('field');
-      return { fieldLink };
+      const calculationLink = conditionLink.forkTo('calculation');
+      const operatorLink = conditionLink.forkTo('operator');
+      const valueLink = conditionLink.forkTo('value');
+      const durationLink = conditionLink.forkTo('duration');
+      return { fieldLink, calculationLink, operatorLink, valueLink, durationLink };
     });
 
     return (
       <Flyout.Container>
         <Flyout.Header>
-          <Flyout.Title>{t(`rulesFlyout.newRule`)}</Flyout.Title>
+          <Flyout.Title>{t(`${tPath}newRule`)}</Flyout.Title>
```

Comment:
I don't think it will be difficult to search. Ex: if you are searching "newRule" string you'll still find it. Otherwise we have to put "rules.flyouts.ruleEditor.xxx" every where.

Reply:
fixed all comments